### PR TITLE
Convert issue and bug templates to use forms

### LINKED ISSUE #10901: Authors to assign component's label
**Is your feature request related to a problem? Please describe.**
Coming from [this](https://github.com/opensearch-project/OpenSearch/issues/10566#issuecomment-1760494890). We need to reduce the time and effort to assign component's label (as we broke it down [here](https://github.com/opensearch-project/OpenSearch/issues/10566)) to newly created issues.

**Describe the solution you'd like**
We need to update the "[feature request](https://github.com/opensearch-project/OpenSearch/blob/main/.github/ISSUE_TEMPLATE/feature_request.md)" and "[bug report](https://github.com/opensearch-project/OpenSearch/blob/main/.github/ISSUE_TEMPLATE/bug_template.md)" templates to include a required dropdown menu that include all component's labels as defined [here](https://github.com/opensearch-project/OpenSearch/issues/10566). Author has the most context regarding the reported bug or the feature request and most probably they will know the related component. Author need to assign at least one value from the dropdown menu, then we need to run a github action that tag the issue with the corresponding github label.

**Additional context**
Using forms in github: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms

### PR DESCRIPTION
**Pull request by mch2**
### Description
This converts bug and feature request templates to forms that require users to select a label.  The added workflow will trigger after the issue cut and apply the actual label to the issue based on the value selected.

This is a continuation of https://github.com/opensearch-project/OpenSearch/pull/11178 cherry picked from @bowenlan-amzn's fork.  

The commit I added sorts the label values and adds a description alerting users to select "Other" if they don't know which label to select.

Tested both on my fork with these issues:
[Feature request](https://github.com/mch2/OpenSearch/issues/270)
[Bug](https://github.com/mch2/OpenSearch/issues/269)

### Related Issues
Resolves #10901 

### Check List
- [x] New functionality includes testing.
  - [ ] ~All tests pass~
- [ ] ~New functionality has been documented.~
  - [ ] ~New functionality has javadoc added~
- [ ] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [ ] ~Commits are signed per the DCO using --signoff~
- [ ] ~Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~
- [ ] ~Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by mch2** [comment]
@msfroh Saw you had started reviewing #11178  - mind taking a look here?

**Comment by kotwanikunal** [comment]
While we are at it - Do you think it makes sense to add in a flaky test form as well?

File: .github/ISSUE_TEMPLATE/bug_template.yml
Comment (anasalkouz) [inline .github/ISSUE_TEMPLATE/bug_template.yml:28]: Nit: may be it will be better if we move the Search label under other search label, to give the chance to user to specify sub-area (e.g. Search:Performance) instead of choosing the generic Search area. Same comment for Indexing

File: .github/ISSUE_TEMPLATE/bug_template.yml
Comment (anasalkouz) [inline .github/ISSUE_TEMPLATE/bug_template.yml:18]: This is a tricky, Since "Build" component is the first option and selected by default. Is there away to not be selected by default. Or make Other as the default option.
Or have something like this 
Screenshot 2023-12-08 at 11 09 40 AM

Comment (mch2) [reply .github/ISSUE_TEMPLATE/bug_template.yml:18]: Was looking at that but didn't see a way to do that yet with GH forms - https://github.com/orgs/community/discussions/61648

File: .github/ISSUE_TEMPLATE/bug_template.yml
Comment (mch2) [reply .github/ISSUE_TEMPLATE/bug_template.yml]: updated

File: .github/ISSUE_TEMPLATE/bug_template.yml
Comment (mch2) [reply .github/ISSUE_TEMPLATE/bug_template.yml:19]: updated to put "Other" first... not perfect but at least if user doesn't make any changes we aren't setting as build.

**Comment by mch2** [comment]
> ❕ Gradle check result for [6bcf739](https://github.com/opensearch-project/OpenSearch/commit/6bcf739a9a746d180dd2d843c5033cc0e23722d2): [UNSTABLE](https://build.ci.opensearch.org/job/gradle-check/31016/)
> 
> * **TEST FAILURES:**
> 
> ```
>       1 org.opensearch.search.SearchWeightedRoutingIT.testMultiGetWithNetworkDisruption_FailOpenEnabled
> ```
> 
> Please review all [flaky tests](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) that succeeded after retry and create an issue if one does not already exist to track the flaky failure.

https://github.com/opensearch-project/OpenSearch/issues/10755

**Comment by mch2** [comment]
> ❌ Gradle check result for [56d4abd](https://github.com/opensearch-project/OpenSearch/commit/56d4abd7588819751293737c86e9dbd98a3adff3): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/31031/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

https://github.com/opensearch-project/OpenSearch/issues/9191
https://github.com/opensearch-project/OpenSearch/issues/7867

**Review by anasalkouz — APPROVED** [review]

**Comment by mch2** [comment]
I forgot I had cut this from main so that i could test it from my fork and reset that branch causing this to get closed.  Should be restored now - @anasalkouz mind taking another look before this is merged?

File: .github/ISSUE_TEMPLATE/bug_template.yml
Comment (peternied) [inline .github/ISSUE_TEMPLATE/bug_template.yml:28]: Could we simplify these to a much smaller list? I think for routing purposes we want each of these options to map to 1 set of owners - who could add the Search: Performance tag when they mark it triaged.

With a larger list it would be easy to constantly adding/removing items from this list based on priorities of the moment.

File: .github/ISSUE_TEMPLATE/bug_template.yml
Comment (peternied) [inline .github/ISSUE_TEMPLATE/bug_template.yml:44]: These 4 bottom boxes take up a lot of web page space, could we consolidate them into a single text box "Additional Details" that has commented out  instructions and hints like including host information or plugins?

File: .github/workflows/triage.yml
Comment (peternied) [inline .github/workflows/triage.yml:1]: What do you think about consolidating this job into the same workflow as https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/add-untriaged.yml

File: .github/ISSUE_TEMPLATE/bug_template.yml
Comment (mch2) [reply .github/ISSUE_TEMPLATE/bug_template.yml:28]: My understanding is that each of these sub categories will have a separate owner no? If this is the case why not skip the extra triaging? Indexing/search could still be selected if the sub is not known.

Comment (anasalkouz) [reply .github/ISSUE_TEMPLATE/bug_template.yml:28]: Yes, I prefer to keep the breakdown for better tracking and ownership. For example: Search: aggregation is totally different area than Search:Remote Search, each one has different code components and different subject matter expert.

File: .github/ISSUE_TEMPLATE/bug_template.yml
Comment (mch2) [reply .github/ISSUE_TEMPLATE/bug_template.yml:44]: Consolidated this - can see my fork for what it looks like - https://github.com/mch2/OpenSearch/issues/new/choose

File: .github/workflows/triage.yml
Comment (mch2) [reply .github/workflows/triage.yml:1]: Updated this in latest commit. Example issue - https://github.com/mch2/OpenSearch/issues/275

**Comment by mch2** [comment]
> ❕ Gradle check result for [56d4abd](https://github.com/opensearch-project/OpenSearch/commit/56d4abd7588819751293737c86e9dbd98a3adff3): [UNSTABLE](https://build.ci.opensearch.org/job/gradle-check/31080/)
> 
> * **TEST FAILURES:**
> 
> ```
>       1 org.opensearch.cluster.allocation.ClusterRerouteIT.testDelayWithALargeAmountOfShards
> ```
> 
> Please review all [flaky tests](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) that succeeded after retry and create an issue if one does not already exist to track the flaky failure.

https://github.com/opensearch-project/OpenSearch/issues/10558

**Comment by mch2** [comment]
> ❌ Gradle check result for [5102640](https://github.com/opensearch-project/OpenSearch/commit/5102640bf7ba0a196e2b04bf86feca736fbef9b0): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/31138/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

https://github.com/opensearch-project/OpenSearch/issues/11581

> ❌ Gradle check result for [f7cdfaf](https://github.com/opensearch-project/OpenSearch/commit/f7cdfafcb2b94dfc14cde096d7a194a58339ebb4): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/31136/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

https://github.com/opensearch-project/OpenSearch/issues/1703

**Comment by mch2** [comment]
> ❌ Gradle check result for [5102640](https://github.com/opensearch-project/OpenSearch/commit/5102640bf7ba0a196e2b04bf86feca736fbef9b0): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/31138/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

https://github.com/opensearch-project/OpenSearch/issues/9191

File: .github/ISSUE_TEMPLATE/bug_template.yml
Comment (peternied) [reply .github/ISSUE_TEMPLATE/bug_template.yml:28]: This list is not for people who know what the different areas are, its for those who _don't_ know; if this list reads like Amazon OpenSearch's organization that sounds like a bug not a feature.

**Review by anasalkouz — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/bug_template.yml
Comment (Pallavi-AWS) [reply .github/ISSUE_TEMPLATE/bug_template.yml:28]: Folks opening issues will have a high level idea on which components it will map to (relevance, indexing, search etc.). For those who are unable to attach a label, we can have the maintainers help with triage.

Comment (mch2) [reply .github/ISSUE_TEMPLATE/bug_template.yml:28]: If we have the extended set of labels as options in the repo I think it should be included here.  I understand the ask to keep it simple, but I still think if we can reduce some manual routing effort we should try it.

Lets keep it as is to start? We can always change if we see it not working - ie. if too many issues are left as "Other"

**Comment by andrross** [comment]
@peternied I see you self-assigned this. Do you want to review/approve it?

**Review by kotwanikunal — APPROVED** [review]

**Review by peternied — APPROVED** [review]

**Comment by peternied** [comment]
> @peternied I see you self-assigned this. Do you want to review/approve it?

Thanks for checking in @andrross - I have been adding myself as 'assigned' to make it easier to track PRs that I'm actively reviewing - but I won't mind if it was merged since I didn't request changes.  🤞 this workflow won't be need after the personalization changes [1] are in.

- [1] https://github.com/opensearch-project/OpenSearch/pull/11600

**Pull request merged**
